src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -106,17 +114,17 @@ func newQemuArch(config HypervisorConfig) (qemuArch, error) {
 		factory = true
 	}
 
-	if config.IOMMU {
-		var q35QemuIOMMUOptions = "accel=kvm,kernel_irqchip=split"
+	// IOMMU and Guest Protection require a split IRQ controller for handling interrupts
+	// otherwise QEMU won't be able to create the kernel irqchip
+	if mp.Type == QemuQ35 && (config.IOMMU || config.GuestProtection) {
```

Comment:
The logic for testing `QemuQ35` here seems wrong. What happens if we have `GuestProtection` but not `Q35`? Is that what should happen?

Reply:
:+1:  thanks, you are right, I didn't test it with `pc`